### PR TITLE
Add support for the GCLOUD_PROJECT environment variable

### LINKED ISSUE #72: Support `GCLOUD_PROJECT` environment variable

### PR DESCRIPTION
**Pull request by jonparrott**
Resolves #72

### CODE REVIEW

File: google/auth/_default.py [hunk 1]
```diff
@@ -254,7 +254,9 @@ def default(request=None):
             If no credentials were found, or if the credentials found were
             invalid.
     """
-    explicit_project_id = os.environ.get(environment_vars.PROJECT)
+    explicit_project_id = os.environ.get(
```
Comment (dhermes) [inline google/auth/_default.py]: Sorry I didn't catch this before. Prefer `os.environ` over `os.getenv`?

Comment (jonparrott) [inline google/auth/_default.py]: I've honestly never used `getenv`.

Comment (jonparrott) [inline google/auth/_default.py]: It seems like a relic.

Comment (dhermes) [inline google/auth/_default.py]: Burn!

**Pull request merged**